internal/rendering/xml-service.go

```diff
@@ -0,0 +1,25 @@
+package xmlservice
+
+import (
+	"encoding/xml"
+)
+
+type XMLWorker interface {
+	Marshal(v interface{}) ([]byte, error)
+}
+
+type XMLWorkerImp struct {}
```

Comment:
if we have separate method `CreateXMLWorker` to create new object, the object definition should be private
```
type xmlWorkerImp struct {}
```